### PR TITLE
refactor(sql): automatically add simple ops implementations

### PR DESCRIPTION
**Pull request by cpcloud**
Follow up to #8338 to clean up the SIMPLE_OPS boilerplate.

### CODE REVIEW

**Comment by cpcloud** [comment]
Looks like I'm on the road to breaking ... everything!

File: ibis/backends/base/sqlglot/compiler.py [hunk 1]
```diff
@@ -192,22 +192,181 @@ class SQLGlotCompiler(abc.ABC):
     UNSUPPORTED_OPERATIONS: frozenset[type[ops.Node]] = frozenset()
     """Set of operations the backend doesn't support."""
 
+    SIMPLE_OPS = {
```
Comment (kszucs) [inline ibis/backends/base/sqlglot/compiler.py:195]: Can you spell out `ops` for consistency with `UNSUPPORTED_OPERATIONS`?

File: ibis/backends/base/sqlglot/compiler.py [hunk 2]
```diff
@@ -192,22 +192,181 @@ class SQLGlotCompiler(abc.ABC):
...
+        ops.IntervalMultiply: sge.Mul,
+        ops.IntervalSubtract: sge.Sub,
+    }
+
     def __init__(self) -> None:
         self.agg = AggGen(aggfunc=self._aggregate)
         self.f = FuncGen()
         self.v = VarGen()
 
     def __init_subclass__(cls, **kwargs):
```
Comment (kszucs) [inline ibis/backends/base/sqlglot/compiler.py:314]: So the only reason we need to construct and attach methods here is because of `has_operation` looks for the method? Couldn't we move this logic into a handler which would compile all value nodes if those are not explicitly overwritten?

File: ibis/backends/base/sqlglot/compiler.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/base/sqlglot/compiler.py:195]: I'd rather do that in a follow up

File: ibis/backends/base/sqlglot/compiler.py (hunk 2 above)
Comment (cpcloud) [reply ibis/backends/base/sqlglot/compiler.py:314]: Maybe, but I don't want to spend time making this improvement perfect here. If you have an idea for a further improvement, that can be done in a follow up.

**Comment by cpcloud** [comment]
Doesn't seem like my CI fixes for label running helped 😞

File: ibis/backends/base/sqlglot/compiler.py [hunk 3]
```diff
@@ -192,22 +206,168 @@ class SQLGlotCompiler(abc.ABC):
...
-            setattr(cls, f"visit_{leaf.__name__}", cls.visit_Undefined)
+            setattr(cls, methodname(op), cls.visit_Undefined)
 
-        super().__init_subclass__(**kwargs)
+        # override existing base class implementations
+        for op, target_name in cls.SIMPLE_OPS.items():
+            setattr(cls, methodname(op), make_impl(op, target_name))
+
+        # add simple ops that are not already implemented
+        for op, target_name in SQLGlotCompiler.SIMPLE_OPS.items():
```
Comment (kszucs) [inline ibis/backends/base/sqlglot/compiler.py:361]: Why is this required?

File: ibis/backends/base/sqlglot/compiler.py [hunk 4]
```diff
@@ -1195,158 +1356,102 @@ def visit_JoinTable(self, op, *, parent, index):
...
+    def visit_GreaterEqual(self, op, *, left, right):
+        return sge.GTE(this=left, expression=right)
+
+    @parenthesize_inputs
+    def visit_Greater(self, op, *, left, right):
+        return sge.GT(this=left, expression=right)
 
-    setattr(SQLGlotCompiler, f"visit_{_op.__name__}", _fmt)
+    @parenthesize_inputs
+    def visit_LessEqual(self, op, *, left, right):
```
Comment (kszucs) [inline ibis/backends/base/sqlglot/compiler.py:1402]: Why are these spelled out instead of being generated like `SIMPLE_OPS`?  `BINARY_INFIX_OPS` could be a mapping.

Comment (cpcloud) [reply ibis/backends/base/sqlglot/compiler.py:1402]: Because implementations call the base class impl. I don't want to keep refactoring this. Is this a blocker?

File: ibis/backends/base/sqlglot/compiler.py (hunk 3 above)
Comment (cpcloud) [reply ibis/backends/base/sqlglot/compiler.py:361]: So that all child classes have these operations.

Comment (kszucs) [reply ibis/backends/base/sqlglot/compiler.py:361]: Ohh, I thought these are already available on the class.

File: ibis/backends/base/sqlglot/compiler.py (hunk 4 above)
Comment (kszucs) [reply ibis/backends/base/sqlglot/compiler.py:1402]: Nope, it is not a blocker.

**Review by kszucs — APPROVED** [review]
